apps/passport-client/components/screens/FrogScreens/FrogManagerScreen.tsx

```diff
@@ -0,0 +1,244 @@
+
+  return (
+    <AppContainer bg="gray">
+      <Container>
+        <textarea onChange={handleFileChange} />
+        {newFrogsError && (
+          <ErrorMessage>Error parsing CSV: {newFrogsError}</ErrorMessage>
+        )}
+
+        {newFrogs.length > 0 && (
+          <>
+            <h2>New Frogs</h2>
+            <DataTable data={newFrogs} />
+            <button onClick={() => refetch(newFrogs)}>Add Frogs</button>
+          </>
+        )}
+
+        <h2>Frogs</h2>
+        {isLoading && <p>Loading...</p>}
+        {error && <ErrorMessage>Error fetching frogs: {error}</ErrorMessage>}
+        {!error && frogs.length === 0 && <p>No frogs</p>}
+        <DataTable data={frogs} />
+      </Container>
+    </AppContainer>
+  );
+}
+
+function useFrogs(): {
+  result: FrogCryptoFrogData[];
+  isLoading: boolean;
+  error: string | undefined;
+  refetch: (frogs: FrogCryptoFrogData[]) => void;
+} {
+  const [result, setResult] = useState<FrogCryptoFrogData[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<string>();
+
+  const identity = useIdentity();
+  const pcds = usePCDCollection();
+  const credentialCache = useCredentialCache();
+
+  const credentialManager = useMemo(
+    () => new CredentialManager(identity, pcds, credentialCache),
+    [credentialCache, identity, pcds]
+  );
+
+  const [pcd, setPcd] = useState<SerializedPCD>();
+  const [inputFrogs, setInputFrogs] = useState<FrogCryptoFrogData[]>([]);
+  useEffect(() => {
+    const fetchPcd = async () => {
+      const pcd = await credentialManager.requestCredential({
+        signatureType: "sempahore-signature-pcd"
+      });
+      setPcd(pcd);
+    };
+    fetchPcd();
+  }, [credentialManager]);
+
+  const req = useMemo(
+    () => ({
+      pcd,
+      frogs: inputFrogs
+    }),
+    [inputFrogs, pcd]
+  );
+
+  useEffect(() => {
+    const abortController = new AbortController();
+
+    const fetchFrogs = async () => {
+      if (!req.pcd) {
+        setError("Waiting for PCD to be ready");
+        return;
+      }
+
+      try {
+        setError(undefined);
+        setIsLoading(true);
+        const result = await requestFrogCryptoManageFrogs(
+          appConfig.zupassServer,
+          req
+        );
+        if (abortController.signal.aborted) {
+          return;
+        }
+        if (result.success) {
+          setResult(result.value.frogs);
+        } else {
+          setError(result.error);
+        }
+      } catch (error) {
+        if (abortController.signal.aborted) {
+          return;
+        }
+        setError(error.message);
+      } finally {
+        if (!abortController.signal.aborted) {
+          setIsLoading(false);
+        }
+      }
+    };
+    fetchFrogs();
+
+    return () => {
+      abortController.abort();
+    };
+  }, [req]);
+
+  return {
+    result,
+    isLoading,
+    error,
+    refetch: setInputFrogs
+  };
+}
+
+function DataTable({ data }: { data: Record<string, any>[] }) {
```

Comment:
worth extracting to a separate file

Reply:
now using a stock component library and contains this screen specific logic